### PR TITLE
Proposal to fix issue #158

### LINKED ISSUE #158: AiiDA-CP2K does not work with Python 3.10
Ubuntu 22.04 upgraded to Python 3.10, which in turn no longer supports `collections.Mapping` and thereby breaks [the tests](https://storage.googleapis.com/cp2k-ci/run-cp2k-aiida-7fb0a63f_report.txt).

https://github.com/aiidateam/aiida-cp2k/blob/0916da4d53ed056a84e25de5e8691841ee23bbab/aiida_cp2k/utils/workchains.py#L32

### PR DESCRIPTION
**Pull request by mkrack**
fixes #158 

Make sure the code is compatible with Python 3.10

### CODE REVIEW

File: aiida_cp2k/utils/workchains.py [hunk 1]
```diff
@@ -27,9 +27,13 @@ def merge_dict(dct, merge_dct):
     :param merge_dct: dct merged into dct (overwrites dct data if in both)
     :return: None
     """
-    import collections
+    # Consider change of the collections interface in Python 3.10 and keep the code backwards compatible
+    try:
+        from collections.abc import Mapping  # pylint: disable=deprecated-class
+    except ImportError:
+        from collections import Mapping  # pylint: disable=deprecated-class
```
Comment (yakutovicha) [inline aiida_cp2k/utils/workchains.py:34]: `collections.abc` was introduced in Python 3.3. I think it is safe to just do this:
```suggestion
    from collections import Mapping
```

**Review by yakutovicha — CHANGES_REQUESTED** [review]
Thanks, @mkrack! I think we can go simple here, see my comment below.

File: aiida_cp2k/utils/workchains.py (hunk 1 above)
Comment (mkrack) [reply aiida_cp2k/utils/workchains.py]: @yakutovicha I guess you mean
```
from collections.abc import Mapping
```

Comment (yakutovicha) [reply aiida_cp2k/utils/workchains.py]: yep, sorry :)

File: aiida_cp2k/utils/workchains.py [hunk 2]
```diff
@@ -27,9 +27,10 @@ def merge_dict(dct, merge_dct):
     :param merge_dct: dct merged into dct (overwrites dct data if in both)
     :return: None
     """
-    import collections
+    # Consider change of the collections interface in Python 3.10
+    from collections.abc import Mapping  # pylint: disable=deprecated-class
```
Comment (yakutovicha) [inline aiida_cp2k/utils/workchains.py:31]: I think pylint disable is not needed, isn't it?

**Comment by mkrack** [comment]
> It looks like the new versions of the checks brought up too many issues. I propose to undo the changes to the `setup.json` file and I fix it in the next PR.

Indeed, the new pylint version detects tons of new issues. Though these changes will certainly improve the code quality, it will be some work to fix all these issues.

**Comment by yakutovicha** [comment]
> > It looks like the new versions of the checks brought up too many issues. I propose to undo the changes to the `setup.json` file and I fix it in the next PR.
> 
> Indeed, the new pylint version detects tons of new issues. Though these changes will certainly improve the code quality, it will be some work to fix all these issues.

yeah, I will address it in a new PR, right after releasing a new version.

File: setup.json [hunk 3]
```diff
@@ -52,5 +52,5 @@
     "license": "MIT License",
     "name": "aiida_cp2k",
     "url": "https://github.com/aiidateam/aiida-cp2k",
-    "version": "1.4.1"
+    "version": "1.4.2"
```
Comment (yakutovicha) [inline setup.json:55]:
```suggestion
    "version": "1.4.1"
```

**Review by yakutovicha — CHANGES_REQUESTED** [review]
let's keep the setup.json intact. I will modify it in a new PR and will release the 1.5.0 version.

File: setup.json [hunk 4]
```diff
@@ -38,8 +38,8 @@
             "coverage"
         ],
         "pre-commit":[
-            "pre-commit~=2.2",
-            "yapf==0.28.0",
+            "pre-commit~=2.19",
+            "yapf==0.32.0",
```
Comment (yakutovicha) [inline setup.json:42]:
```suggestion
            "pre-commit~=2.2",
            "yapf==0.28.0",
```

**Review by yakutovicha — APPROVED** [review]
Looks good to me!

**Comment by mkrack** [comment]
@yakutovicha Thank you for your support. We are fine with a new aiida_cp2k sub-release which will work with the upcoming CP2K release end of June also with Python 3.10.

**Comment by yakutovicha** [comment]
@mkrack I've just released a new version of the plugin: https://github.com/aiidateam/aiida-cp2k/releases/tag/v1.5.0

**Pull request merged**